### PR TITLE
1.2.0 dev

### PR DESCRIPTION
**Pull request by Sauilitired**

### CODE REVIEW

**Comment by Sauilitired** [comment]
Draft until #157 is fixed.

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +59,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
  
 ### Fixed
  - Fix arguments with no required children not being executors (cloud-brigadier)
+ - Detect and throw an exception for ambiguous nodes in more cases
+ - CloudBrigadierManager no longer forgets the command sender
```
Comment (Sauilitired) [inline CHANGELOG.md:63]: these ended up in the wrong section 😆

File: cloud-annotations/src/main/java/cloud/commandframework/annotations/AnnotationParser.java [hunk 2]
```diff
@@ -336,6 +371,19 @@ public AnnotationParser(
         return commands;
     }
 
+    private @NonNull SyntaxFragment findSyntaxFragment(
+            final @NonNull List<@NonNull SyntaxFragment> fragments,
+            final @NonNull String argumentName
+    ) {
+        for (SyntaxFragment fragment : fragments) {
```
Comment (Sauilitired) [inline cloud-annotations/src/main/java/cloud/commandframework/annotations/AnnotationParser.java:378]:
```suggestion
        for (final SyntaxFragment fragment : fragments) {
```

File: cloud-annotations/src/main/java/cloud/commandframework/annotations/MethodCommandExecutionHandler.java [hunk 3]
```diff
@@ -82,19 +90,36 @@ public void execute(final @NonNull CommandContext<C> commandContext) {
...
+                        if (value != null) {
+                           break;
+                        }
+                    }
+                    if (value != null) {
+                        arguments.add(value);
+                    } else {
+                        throw new IllegalArgumentException(String.format(
+                                "Unknown command parameter '%s' in method '%s'",
+                                parameter.getName(), this.methodHandle.toString()
```
Comment (Sauilitired) [inline cloud-annotations/src/main/java/cloud/commandframework/annotations/MethodCommandExecutionHandler.java:109]:
```suggestion
                                parameter.getName(),
                                this.methodHandle.toString()
```

File: cloud-annotations/src/main/java/cloud/commandframework/annotations/injection/RawArgs.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Used to inject {@link cloud.commandframework.context.CommandContext#getRawInput()} into a
+ * {@link cloud.commandframework.annotations.CommandMethod}
+ * <p>
+ * This should only be used on {@code String[]}
```
Comment (Sauilitired) [inline cloud-annotations/src/main/java/cloud/commandframework/annotations/injection/RawArgs.java:35]:
```suggestion
 * This should only be used on {@code String[]}
 *
```

File: cloud-core/src/main/java/cloud/commandframework/CommandManager.java [hunk 5]
```diff
@@ -781,11 +854,60 @@ public void setSetting(
...
+         * @since 1.2.0
+         */
+        OVERRIDE_EXISTING_COMMANDS
+    }
+
+    /**
+     * The point in the registration lifecycle for this commands manager
+     *
+     * @since 1.2.0
+     */
```
Comment (Sauilitired) [inline cloud-core/src/main/java/cloud/commandframework/CommandManager.java:887]: look over javadoc formatting (really minor)

**Review by Sauilitired — COMMENTED** [review]
I can just find really minor formatting issues. Before merging, all versions will need to be changed from `1.2.0-SNAPSHOT` to `1.2.0`

**Review by broccolai — APPROVED** [review]

**Review by Proximyst — APPROVED** [review]
stop bullying me

**Pull request merged**